src/shared/components/data-table/DataTable.test.tsx

```diff
@@ -0,0 +1,121 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { times } from 'lodash';
+
+import DataTable from './DataTable';
+import { defaultDataTableState } from 'src/shared/components/data-table/dataTableReducer';
+
+import { type TableProps } from 'src/shared/components/data-table/DataTable';
+
+import { TableData, DataTableColumns } from './dataTableTypes';
+
+export const createDataTableSampleData = (
+  rows: number,
+  columns: number
+): { data: TableData; columns: DataTableColumns } => {
+  return {
+    data: times(rows, (row) =>
+      times(columns, (column) => `Cell ${row},${column}`)
+    ),
+    columns: times(columns, (column) => ({
+      columnId: `${column}`,
+      title: `Column ${column}`,
+      helpText: `Column ${column} help text`,
+      isSortable: true
+    }))
+  };
+};
+
+const { columns, data } = createDataTableSampleData(15, 10);
+
+const defaultProps = {
+  state: {
+    ...defaultDataTableState,
+    data
+  },
+  columns
+};
+
+describe('<DataTable />', () => {
+  const renderDataTable = (props: Partial<TableProps> = {}) =>
+    render(<DataTable {...defaultProps} {...props} />);
+
+  let container: HTMLElement;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('renders without error', () => {
+    container = renderDataTable().container;
+    expect(() => container).not.toThrow();
+  });
+
+  it('displays correct number of rows per page', () => {
+    container = renderDataTable({
+      state: {
+        ...defaultProps.state,
+        rowsPerPage: 10
+      }
+    }).container;
+
+    expect(container.querySelectorAll('tbody tr').length).toBe(10);
+
+    container = renderDataTable({
+      state: {
+        ...defaultProps.state,
+        rowsPerPage: 20
+      }
+    }).container;
+
+    expect(container.querySelectorAll('tbody tr').length).toBe(
+      defaultProps.state.data.length
+    );
+
+    container = renderDataTable({
+      state: {
+        ...defaultProps.state,
+        rowsPerPage: Infinity
+      }
+    }).container;
+
+    expect(container.querySelectorAll('tbody tr').length).toBe(
+      defaultProps.state.data.length
+    );
```

Comment:
No difference from the check above?

Reply:
Here we check if the correct number of rows are displayed after setting `rowsPerPage` to `Infinity` (it was `20` in the previous check)

Reply:
I know 😄 

- We start with 10, right? Do you want to skip `rowsPerPage` in one of the checks to see the default behaviour?
- The behaviour in the Infinity check is no different from the behaviour in the 20 rows per page check. Perhaps generate the number of rows that is above the maximum number offered by the UI (over 100), and make sure all rows are present?

Reply:
I've updated it to cover the following scenarios:

Total rows: 15, rowsPerPage: default, Displayed rows should be: 10
Total rows: 15, rowsPerPage: Infinity, Displayed rows should be: 15
Total rows: 15, rowsPerPage: default, currentPageNumber:2, Displayed rows should be: 10